ksql-execution/src/main/java/io/confluent/ksql/execution/expression/formatter/ExpressionFormatter.java

```diff
@@ -153,11 +153,16 @@ public String visitDereferenceExpression(
         final Context context) {
       final String baseString = process(node.getBase(), context);
       if (node.getBase() instanceof QualifiedNameReference) {
-        return baseString + KsqlConstants.DOT + formatIdentifier(node.getFieldName());
+        return baseString + KsqlConstants.DOT
+            + quoteReservedWord(formatIdentifier(node.getFieldName()), context);
```

Comment:
why are we only quoting reserved here and not elsewhere? why not just do it in `formatIdentifier`?